### ISSUE TITLE
Add support for Java Records (JEP 395)

### ISSUE THREAD

**Issue by simon04** [quicktype/quicktype#2019]
Please support Java Records as an alternative output mode (similar to "Use lombok").

Java 16 introduced Java Records:

- JEP 395: Records – https://openjdk.org/jeps/395
- https://www.baeldung.com/java-record-keyword

The welcome example from https://app.quicktype.io/ should output

```java
package io.quicktype;

public record Welcome (
    String greeting,
    String[] instructions) {
}
```

The multiple JSON example should output

```java
// Album.java

package io.quicktype;

public record Album (
    String name,
    Artist artist,
    Track[] tracks) {
}

// Artist.java

package io.quicktype;

public record Artist (
    String name,
    long founded,
    String[] members) {
}

// Track.java

package io.quicktype;

public record Track (
    String name,
    long duration) {
}
```

Thank you for maintaining this great tool!

**Comment by dvdsgl** [comment]
Very cool improvement to Java! Please consider sending a PR for this.

**Comment by dvdsgl** [comment]
GitHub closed this accidentally when it prematurely merged #2165
